### ISSUE TITLE
Issue with installation using conda involving pytest

### ISSUE THREAD

**Issue by fabianrost84** [Jupyter-contrib/jupyter_nbextensions_configurator#55]
When trying to install jupyter_nbextensions_configurator with conda I get the following error:
```
>conda install -c conda-forge jupyter_nbextensions_configurator
Fetching package metadata .................
Solving package specifications: .

Package plan for installation in environment /home/fabrost/miniconda3:

The following NEW packages will be INSTALLED:

    bleach:                            2.0.0-py36_0          conda-forge
    decorator:                         4.1.2-py36_0          conda-forge
    entrypoints:                       0.2.3-py36_1          conda-forge
    gmp:                               6.1.2-0               conda-forge
    html5lib:                          1.0.1-py_0            conda-forge
    ipykernel:                         4.7.0-py36_0          conda-forge
    ipython:                           6.2.1-py36_0          conda-forge
    ipython_genutils:                  0.2.0-py36_0          conda-forge
    jedi:                              0.10.2-py36_0         conda-forge
    jinja2:                            2.10-py36_0           conda-forge
    jsonschema:                        2.6.0-py36_0          conda-forge
    jupyter_client:                    5.2.0-py36_0          conda-forge
    jupyter_contrib_core:              0.3.3-py36_0          conda-forge
    jupyter_core:                      4.4.0-py_0            conda-forge
    jupyter_nbextensions_configurator: 0.2.8-py36_0          conda-forge
    libsodium:                         1.0.15-1              conda-forge
    markupsafe:                        1.0-py36_0            conda-forge
    mistune:                           0.8.3-py_0            conda-forge
    nbconvert:                         5.3.1-py_1            conda-forge
    nbformat:                          4.4.0-py36_0          conda-forge
    notebook:                          5.2.2-py36_1          conda-forge
    pandoc:                            2.0.5-0               conda-forge
    pandocfilters:                     1.4.1-py36_0          conda-forge
    pexpect:                           4.3.1-py36_0          conda-forge
    pickleshare:                       0.7.4-py36_0          conda-forge
    prompt_toolkit:                    1.0.15-py36_0         conda-forge
    ptyprocess:                        0.5.2-py36_0          conda-forge
    pygments:                          2.2.0-py36_0          conda-forge
    python-dateutil:                   2.6.1-py36_0          conda-forge
    pyyaml:                            3.12-py36_1           conda-forge
    pyzmq:                             16.0.2-py36_2         conda-forge
    simplegeneric:                     0.8.1-py36_0          conda-forge
    terminado:                         0.8.1-py36_0          conda-forge
    testpath:                          0.3.1-py36_0          conda-forge
    tornado:                           4.5.2-py36_0          conda-forge
    traitlets:                         4.3.2-py36_0          conda-forge
    wcwidth:                           0.1.7-py36_0          conda-forge
    webencodings:                      0.5-py36_0            conda-forge
    zeromq:                            4.2.1-1               conda-forge

The following packages will be SUPERSEDED by a higher-priority channel:

    conda:                             4.3.30-py36h5d9f9f4_0             --> 4.3.29-py36_0 conda-forge
    conda-env:                         2.6.0-h36134e3_1                  --> 2.6.0-0       conda-forge

Proceed ([y]/n)? 

+ /home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator enable --sys-prefix
Traceback (most recent call last):
  File "/home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3138, in <module>
    @_call_aside
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3122, in _call_aside
    f(*args, **kwargs)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3151, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 664, in _build_master
    ws.require(__requires__)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 981, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pytest; python_version >= "3.4" or python_version == "2.7" and extra == "test"' distribution was not found and is required by jupyter-client

ERROR conda.core.link:_execute_actions(337): An error occurred while installing package 'conda-forge::jupyter_nbextensions_configurator-0.2.8-py36_0'.
LinkError: post-link script failed for package conda-forge::jupyter_nbextensions_configurator-0.2.8-py36_0
running your command again with `-v` will provide additional information
location of failed script: /home/fabrost/miniconda3/bin/.jupyter_nbextensions_configurator-post-link.sh
==> script messages <==
+ /home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator enable --sys-prefix
Traceback (most recent call last):
  File "/home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3138, in <module>
    @_call_aside
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3122, in _call_aside
    f(*args, **kwargs)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3151, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 664, in _build_master
    ws.require(__requires__)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 981, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pytest; python_version >= "3.4" or python_version == "2.7" and extra == "test"' distribution was not found and is required by jupyter-client


Attempting to roll back.


LinkError: post-link script failed for package conda-forge::jupyter_nbextensions_configurator-0.2.8-py36_0
running your command again with `-v` will provide additional information
location of failed script: /home/fabrost/miniconda3/bin/.jupyter_nbextensions_configurator-post-link.sh
==> script messages <==
+ /home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator enable --sys-prefix
Traceback (most recent call last):
  File "/home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3138, in <module>
    @_call_aside
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3122, in _call_aside
    f(*args, **kwargs)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3151, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 664, in _build_master
    ws.require(__requires__)
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 981, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/fabrost/miniconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 867, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pytest; python_version >= "3.4" or python_version == "2.7" and extra == "test"' distribution was not found and is required by jupyter-client
```
Installing `pytest` as well works around the issue:

```
> conda install -c conda-forge jupyter_nbextensions_configurator pytest
Fetching package metadata .................
Solving package specifications: .

Package plan for installation in environment /home/fabrost/miniconda3:

The following NEW packages will be INSTALLED:

    attrs:                             17.3.0-py_0           conda-forge
    bleach:                            2.0.0-py36_0          conda-forge
    decorator:                         4.1.2-py36_0          conda-forge
    entrypoints:                       0.2.3-py36_1          conda-forge
    gmp:                               6.1.2-0               conda-forge
    html5lib:                          1.0.1-py_0            conda-forge
    ipykernel:                         4.7.0-py36_0          conda-forge
    ipython:                           6.2.1-py36_0          conda-forge
    ipython_genutils:                  0.2.0-py36_0          conda-forge
    jedi:                              0.10.2-py36_0         conda-forge
    jinja2:                            2.10-py36_0           conda-forge
    jsonschema:                        2.6.0-py36_0          conda-forge
    jupyter_client:                    5.2.0-py36_0          conda-forge
    jupyter_contrib_core:              0.3.3-py36_0          conda-forge
    jupyter_core:                      4.4.0-py_0            conda-forge
    jupyter_nbextensions_configurator: 0.2.8-py36_0          conda-forge
    libsodium:                         1.0.15-1              conda-forge
    markupsafe:                        1.0-py36_0            conda-forge
    mistune:                           0.8.3-py_0            conda-forge
    nbconvert:                         5.3.1-py_1            conda-forge
    nbformat:                          4.4.0-py36_0          conda-forge
    notebook:                          5.2.2-py36_1          conda-forge
    pandoc:                            2.0.5-0               conda-forge
    pandocfilters:                     1.4.1-py36_0          conda-forge
    pexpect:                           4.3.1-py36_0          conda-forge
    pickleshare:                       0.7.4-py36_0          conda-forge
    pluggy:                            0.6.0-py_0            conda-forge
    prompt_toolkit:                    1.0.15-py36_0         conda-forge
    ptyprocess:                        0.5.2-py36_0          conda-forge
    py:                                1.5.2-py_0            conda-forge
    pygments:                          2.2.0-py36_0          conda-forge
    pytest:                            3.3.1-py36_0          conda-forge
    python-dateutil:                   2.6.1-py36_0          conda-forge
    pyyaml:                            3.12-py36_1           conda-forge
    pyzmq:                             16.0.2-py36_2         conda-forge
    simplegeneric:                     0.8.1-py36_0          conda-forge
    terminado:                         0.8.1-py36_0          conda-forge
    testpath:                          0.3.1-py36_0          conda-forge
    tornado:                           4.5.2-py36_0          conda-forge
    traitlets:                         4.3.2-py36_0          conda-forge
    wcwidth:                           0.1.7-py36_0          conda-forge
    webencodings:                      0.5-py36_0            conda-forge
    zeromq:                            4.2.1-1               conda-forge

The following packages will be SUPERSEDED by a higher-priority channel:

    conda:                             4.3.30-py36h5d9f9f4_0             --> 4.3.29-py36_0 conda-forge
    conda-env:                         2.6.0-h36134e3_1                  --> 2.6.0-0       conda-forge

Proceed ([y]/n)? 

+ /home/fabrost/miniconda3/bin/jupyter-nbextensions_configurator enable --sys-prefix
Enabling: jupyter_nbextensions_configurator
- Writing config: /home/fabrost/miniconda3/etc/jupyter
    - Validating...
      jupyter_nbextensions_configurator  OK
Enabling notebook nbextension nbextensions_configurator/config_menu/main...
Enabling tree nbextension nbextensions_configurator/tree_tab/main...
```

**Comment by jcb91** [comment]
Hmm, weird. I've seen this once before somewhere, but I forget where. This appears to be an issue either with `jupyter_client`, or with dependency resolution, rather than this repo directly, since it's occurring in a call from the shell script generated by setuptools. It seems like `pkg_resources` is attempting to get the optional-extra `test` dependencies of `jupyter_client`, but I don't follow why that would be the case :thinking:

**Comment by jcb91** [comment]
Found the previous instance: https://github.com/conda-forge/jupyter_nbextensions_configurator-feedstock/issues/21

**Comment by jcb91** [comment]
should be fixed by https://github.com/Jupyter-contrib/jupyter_nbextensions_configurator/pull/56, which is included in the [0.3.0 conda-forge release](https://anaconda.org/conda-forge/jupyter_nbextensions_configurator/files?version=0.3.0)
